app/jobs/email_delivery_jobs/base.rb

```diff
@@ -0,0 +1,69 @@
+module EmailDeliveryJobs
+  class Base < ActiveJob::Base
+
+    #
+    # Send a single email to a recipient informing them about a petition that they have signed
+    # Implemented as a custom job rather than using action mailers #deliver_later so we can do
+    # extra checking before sending the email
+    #
+
+    queue_as :default
+
+    def perform(signature:, timestamp_name:, petition:,
+      requested_at_as_string:, mailer: PetitionMailer.name, logger: nil)
+
+      @mailer = mailer.constantize
+      @signature = signature
+      @petition = petition
+      @requested_at = requested_at_as_string.in_time_zone
+      @timestamp_name = timestamp_name
+      @logger = logger
+
+      if petition_has_not_been_updated?
```

Comment:
I think we can make this job idempotent if we made it also check the receipt timestamp for the signature and return if the stamp is not `petition_timestamp`.  The enqueue job already filters them out via the `petition.signatures_to_email_for` call, but we can do a check here too.